src/openms/source/APPLICATIONS/TOPPBase.cpp

```diff
@@ -297,7 +298,17 @@ namespace OpenMS
         {
           in_ini = param_cmdline_.getValue("ini");
           Param ini_params;
-          ParamXMLFile().load(in_ini.toString(), ini_params);
+          std::string in_ini_path = in_ini.toString();
+          if (FileHandler::hasValidExtension(in_ini_path, FileTypes::Type::JSON))
+          {
+            // CWL doesn't carry information about the tree structure,
+            // so we fill it in with the default values to have access to this information
+            param_inifile_ = this->getDefaultParameters_();
```

Comment:
note sure you need this here, since `ParamCWLFile().load(in_ini_path, ini_params);` does not depend on `param_inifile_`

Reply:
Nicely caught, that was a type!